pkg/apis/imageregistry/v1/types.go

```diff
@@ -150,10 +150,21 @@ type ImageRegistryConfigProxy struct {
 }
 
 type ImageRegistryConfigStorageS3 struct {
-	Bucket         string `json:"bucket,omitempty"`
-	Region         string `json:"region,omitempty"`
+	// Bucket is the bucket name in which you want to store the registry's data
+	// Optional, will be generated if not provided
+	Bucket string `json:"bucket,omitempty"`
+	// Region is the AWS region in which your bucket exists
+	// Optional, will be set based on the installed AWS Region
+	Region string `json:"region,omitempty"`
+	// RegionEndpoint is the endpoint for S3 compatible storage services
+	// Optional, defaults based on the Region that is provided
 	RegionEndpoint string `json:"regionEndpoint,omitempty"`
-	Encrypt        bool   `json:"encrypt,omitempty"`
+	// Encrypt specifies whether the registry stores the image in encrypted format or not
+	// Optional, defaults to false
+	Encrypt bool `json:"encrypt,omitempty"`
+	// KeyID is the KMS key ID to use for encryption
+	// Optional, Encrypt must be true, or this parameter is ignored
+	KeyID string `json:"keyID,omitempty"`
```

Comment:
what is the behavior if someone changes Encrypt or KeyID later?

Reply:
Ultimately, we would want the syncStorage to pick up the change and re-deploy the image-registry with the new env vars

Reply:
so today the behavior is.....?  i just want to make sure someone won't end up in a bad state if they set this after the fact (where we might end up half-configuring the system, ie setting up the registry config but not configuring the bucket properly).

What's going to happen when we see that spec storage config doesn't match status storage config currently?

Reply:
@coreydaley bump ^^

Reply:
I'm working on this, it may be part of a larger change to deal with what should happen when things change like this.

Reply:
"if i add a keyID to the custom resource, it changes the encryption on the bucket,and if i remove it, it changes it back "

i take it that's the answer to my question on this?

Reply:
Yes, that is the answer now :) I'm not sure what it was before ...